Added additional Note Control options

moogmusic opened the pull request:
I'm not suggesting this be merged into the code but I wondered if you could check over my coding and understanding of what's going on before I try to install it onto my Carbon.

Thanks

Added additional Note Control options for Roland PK-5 Sound Effects Factory Set as follows:

Pedal 0: Mute/unmute track 1
Pedal 2: Mute/unmute track 2
Pedal 4: Mute/unmute track 3
Pedal 5: Mute/unmute track 4
Pedal 7: Mute/unmute track 5
Pedal 9: Mute/unmute track 6
Pedal C: Reset then run/stop playback.

Pedal/MIDI assignments source: https://static.roland.com/assets/media/pdf/PK-5_OM.pdf

moogmusic committed:
Added additional Note Control options

Added additional Note Control options for Roland PK-5 Sound Effects Factory Set as follows:

Pedal 0: Mute/unmute track 1
Pedal 2: Mute/unmute track 2
Pedal 4: Mute/unmute track 3
Pedal 5: Mute/unmute track 4
Pedal 7: Mute/unmute track 5
Pedal 9: Mute/unmute track 6
Pedal C: Reset then run/stop playback.

kilpatrickaudio commented:
This looks cool. I might merge these in for the next update. I don't mind custom stuff like this as long as it doesn't add menu items most people won't use. But to have extra non-conflicting mappings seems totally fine.

moogmusic commented:
Cool - I'll have to think of something to do with the other six pedals then...

moogmusic committed:
Added MIDI Tap Tempo

Added additional Note Control options for Roland PK-5 Sound Effects Factory Set as follows:

Pedal B: MIDI Tap Tempo

Pedal/MIDI assignments source: https://static.roland.com/assets/media/pdf/PK-5_OM.pdf

moogmusic committed:
Started adding in final pedal controls

moogmusic committed:
Finished adding specific actions for each pedal

Added additional Note Control options for Roland PK-5 Sound Effects Factory Set as follows:

Pedal 0: Mute/unmute track 1
Pedal 1: Select track 1
Pedal 2: Mute/unmute track 2
Pedal 3: Select track 2
Pedal 4: Mute/unmute track 3
Pedal 5: Mute/unmute track 4
Pedal 6: Arp on/off track 2
Pedal 7: Mute/unmute track 5
Pedal 8:Arp on/off track 3
Pedal 9: Mute/unmute track 6
Pedal A: Arp on/off track 4
Pedal B: Tap tempo
Pedal C: Reset then run/stop playback.

Pedal/MIDI assignments source: https://static.roland.com/assets/media/pdf/PK-5_OM.pdf

moogmusic commented:
I've finished adding in controls for all the other pedals - these are quite specific to how I intend to use Carbon so I'm not sure it would be useful for other users. I've tried to compile the code myself on Ubuntu but I can't figure out all the dependencies, do you have list somewhere I can just apt-get install?

Thanks

Added additional Note Control options for Roland PK-5 Sound Effects Factory Set as follows:

Pedal 0: Mute/unmute track 1
Pedal 1: Select track 1
Pedal 2: Mute/unmute track 2
Pedal 3: Select track 2
Pedal 4: Mute/unmute track 3
Pedal 5: Mute/unmute track 4
Pedal 6: Arp on/off track 2
Pedal 7: Mute/unmute track 5
Pedal 8:Arp on/off track 3
Pedal 9: Mute/unmute track 6
Pedal A: Arp on/off track 4
Pedal B: Tap tempo
Pedal C: Reset then run/stop playback.

Pedal/MIDI assignments source: https://static.roland.com/assets/media/pdf/PK-5_OM.pdf

kilpatrickaudio commented:
This is cool, thanks! As for building please see the file called BUILDING in the project root. If you want to test the code easily you should get a JTAG programmer and add a connector onto the X10 location. It follows the standard 20 pin ARM JTAG pinout supported by devices like the STLink and JLink. If you have specific problems that are blocking you, I can help you, but in generally it's up to you to set up your environment and figure out the tools.
